packages/TorneloScoresheet/src/hooks/appMode/graphicalRecordingState.ts

```diff
@@ -1,19 +1,109 @@
+import { chessEngine } from '../../chessEngine/chessEngineInterface';
 import {
   AppMode,
   AppModeState,
   GraphicalRecordingMode,
 } from '../../types/AppModeState';
+import { ChessBoardPositions } from '../../types/ChessBoardPositions';
+import { ChessMove, MovePlie, PliePositions } from '../../types/ChessMove';
 
 type GraphicalRecordingStateHookType = [
   GraphicalRecordingMode,
   {
     goToEndGame: () => void;
     goToTextInput: () => void;
     goToArbiterMode: () => void;
+    move: (pliePosition: PliePositions) => void;
   },
 ];
 
+/**
+ * Process Plie based on the from and to positions
+ * Returns the next Plie with it's starting fen
+ * Returns null if move was impossible
+ * @param currentPlie the Plie that should be processed
+ * @returns next plie if move is successfull else null
+ */
+const processPlie = (currentPlie: MovePlie): MovePlie | null => {
+  if (currentPlie.from === undefined || currentPlie.to === undefined) {
+    return null;
+  }
+  const nextFen = chessEngine.makeMove(currentPlie.startingFen, {
+    from: currentPlie.from,
+    to: currentPlie.to,
+  });
+
+  // if move was not possible, return null
+  if (nextFen === null) {
+    return null;
+  }
+
+  // create next plie
+  const nextPlie: MovePlie = { startingFen: nextFen };
+
+  return nextPlie;
+};
+
+/**
+ * Processes a player's move give from and to positons
+ * Will Return the next board state and the move history array
+ * Will return null if move is imporssible
+ * @param fromPos postion piece starts from
+ * @param toPos postion piece goes to
+ * @param appModeState the GraphicalRecordingMode state
+ * @param setAppModeStateAfterMove the state update function
+ * @returns [boardPositions, moveHistory] or null
+ */
+const processPlayerMove = (
+  pliePositions: PliePositions,
+  moveHistory: ChessMove[],
+): [ChessBoardPositions, ChessMove[]] | null => {
+  const currentMoveIdx = moveHistory.length - 1;
+  const currentMove = moveHistory[currentMoveIdx];
+
+  // White's move
+  if (currentMove.blackMove === undefined) {
+    // perform move
+    currentMove.whiteMove.from = pliePositions.from;
+    currentMove.whiteMove.to = pliePositions.to;
+    const nextPlie = processPlie(currentMove.whiteMove);
+    if (nextPlie === null) {
+      return null;
+    }
+
+    // return new state
+    currentMove.blackMove = nextPlie;
+    moveHistory.splice(currentMoveIdx);
+    return [
+      chessEngine.fenToBoardPositions(nextPlie.startingFen),
+      [...moveHistory, currentMove],
+    ];
+  }
+
+  // Black's move
+  else {
+    // perform move
+    currentMove.blackMove.from = pliePositions.from;
+    currentMove.blackMove.to = pliePositions.to;
+    const nextPlie = processPlie(currentMove.blackMove);
+    if (nextPlie === null) {
+      return null;
+    }
+
+    // return new state
+    const nextMove: ChessMove = {
+      moveNo: currentMove.moveNo + 1,
+      whiteMove: nextPlie,
+    };
+    moveHistory.splice(currentMoveIdx);
```

Comment:
Also - don't think this if/else is necessary. Can't we use:
```ts
const moveIndex = currentMove.blackMove === undefined ? 'whiteMove' : 'blackMove';
currentMove[moveIndex].from = pliePositions.from;
...
```

I think the code would be much clearer without the massive if/else

Reply:
You may need to use `'whiteMove' as const` and `'blackMove' as const` or something similar

Reply:
yeh left it like that so it's clear what the difference between a white move update and a black move update is, but ive chagned it now i think its even clearer